src/promise/Promise.php

```diff
@@ -23,10 +23,11 @@
 
 namespace pocketmine\promise;
 
+use function count;
 use function spl_object_id;
 
 /**
- * @phpstan-template TValue
+ * @phpstan-template-covariant TValue
  */
 final class Promise{
```

Comment:
So are we now determined to turn `Promise` into a general-purpose async util class?